### PR TITLE
CoreOS fix

### LINKED ISSUE #532: DISTRO_ID should be used instead of DISTRO_NAME
Right now, a lot of the functionality switches implementations based on a mangled version of the [NAME](https://www.freedesktop.org/software/systemd/man/os-release.html#NAME=) field from `/etc/os-release`. This isn't safe to do because this field is intended for consumption by the user. The correct field to use is the [ID](https://www.freedesktop.org/software/systemd/man/os-release.html#ID=) field. This is actually meant for consumption by scripts and OS vendors will make an effort not to screw with this too often.

### PR DESCRIPTION
**Pull request by hglkrijger**
- minor cleanup
- add `distro_code_name` which is populated from the `ID` field
- detect CoreOS via `distro_code_name`
- fixes #532 

/cc @jinhyunr @brendandixon @crawford

### CODE REVIEW

File: azurelinuxagent/common/osutil/factory.py [hunk 1]
```diff
@@ -17,64 +17,73 @@
...
     elif distro_name == "freebsd":
         return FreeBSDOSUtil()
+
     elif distro_name == "bigip":
         return BigIpOSUtil()
+
     else:
-        logger.warn("Unable to load distro implementation for {0}.", distro_name)
-        logger.warn("Use default distro implementation instead.")
+        logger.warn("Unable to load distro implementation for {0}. Using "
```
Comment (jinhyunr) [inline azurelinuxagent/common/osutil/factory.py]: Does this work? I've seen it failing sometimes and had to use %s instead.. :(

File: azurelinuxagent/common/osutil/factory.py [hunk 2]
```diff
@@ -17,64 +17,73 @@
...
         return CoreOSUtil()
+
     if distro_name == "suse":
-        if distro_full_name=='SUSE Linux Enterprise Server' and \
-           Version(distro_version) < Version('12') or \
-           distro_full_name == 'openSUSE' and \
-           Version(distro_version) < Version('13.2'):
+        if distro_full_name == 'SUSE Linux Enterprise Server' \
+                and Version(distro_version) < Version('12') \
+                or distro_full_name == 'openSUSE' and Version(distro_version) < Version('13.2'):
```
Comment (jinhyunr) [inline azurelinuxagent/common/osutil/factory.py]: Any reason for this format?

Comment (hglkrijger) [inline azurelinuxagent/common/osutil/factory.py]: No, just easier to read.

File: azurelinuxagent/common/osutil/factory.py (hunk 1 above)
Comment (hglkrijger) [inline azurelinuxagent/common/osutil/factory.py]: Yes, our logging class supports this behavior. Note the usage is unchanged, just combining the logs into one line.

**Pull request merged**